pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -176,14 +177,280 @@ func TestPriorityQueue_AddWithReversePriorityLessFunc(t *testing.T) {
 	}
 }
 
+func clusterEventsToList(clusterEvents []*clusterEvent) *list.List {
+	l := list.New()
+	for _, event := range clusterEvents {
+		l.PushBack(event)
+	}
+	return l
+}
+
+func listToClusterEvents(l *list.List) []*clusterEvent {
+	clusterEvents := []*clusterEvent{}
+	for e := l.Front(); e != nil; e = e.Next() {
+		clusterEvents = append(clusterEvents, e.Value.(*clusterEvent))
+	}
+	return clusterEvents
+}
+
+func TestDone(t *testing.T) {
```

Comment:
I wonder if you could have a test table for both this and the next test.

As input, you can have a list of actions:
- pod enqueued
- pod popped
- event

then you check `wantInflightPods`, `wantActiveQ`, `wantBackoff`, `wantUnschedulable`.

Reply:
How about like this https://github.com/kubernetes/kubernetes/pull/118438/commits/34c8a5b0eaa021e9e5735a776854f095e2cdf706